src/module-elasticsuite-virtual-category/Plugin/Widget/ProductsListPlugin.php

```diff
@@ -94,10 +94,11 @@ public function afterCreateCollection(ProductsList $subject, $collection)
         $storeId = $this->storeManager->getStore()->getId();
 
         if ($subject->getData('condition_option') == 'condition' || !$subject->getData('condition_option')) {
-            $conditions = $subject->getData('conditions_encoded') ?: $subject->getData('conditions');
-            if ($conditions) {
-                $conditions = $this->conditionsHelper->decode($conditions);
+            if (!$conditions = $subject->getData('conditions_encoded') ?: $subject->getData('conditions')) {
+                return $collection;
             }
+
+            $conditions = $this->conditionsHelper->decode($conditions);
             foreach ($conditions as $condition) {
```

Comment:
I think your solution is over-complicated.

I'd prefer something like this (with just one guard condition) : 

```php
$conditions = $subject->getData('conditions_encoded') ?: $subject->getData('conditions');

if ($conditions) {
    $conditions = $this->conditionsHelper->decode($conditions);
    foreach ($conditions as $condition) {
    ... legacy ...
    }    
}

return $collection
```